objcgen/NameGenerator.cs

```diff
@@ -62,10 +62,11 @@ public static string GetTypeName (Type t)
 			if (t.IsByRef) {
 				var et = t.GetElementType ();
 				var typecode = Type.GetTypeCode (et);
+
 				if (typecode == TypeCode.Decimal || typecode == TypeCode.DateTime) // This is boxed into NSDecimalNumber/NSDate
 					return GetTypeName (et) + "_Nonnull * _Nullable";
 
-				return GetTypeName (et) + (et.IsValueType ? " " : " _Nonnull ") + "* _Nullable";
+				return GetTypeName (et) + " * " + (et.IsValueType ? "_Nullable" : "_Nonnull");
```

Comment:
there are two levels here and it seems you removed one
i.e. it's a pointer to a pointer and the nullability of each can differ (just live two lines above)